src/common/post.ts

```diff
@@ -1104,3 +1328,46 @@ export const ensurePostAnalyticsPermissions = async ({
     );
   }
 };
+
+export const createFreeformPost = async (
+  ctx: AuthContext,
+  args: CreatePostArgs,
+  options?: Partial<{ entityManager: EntityManager }>,
+) => {
+  const { sourceId, image } = args;
+  const { con, userId } = ctx;
+  const id = await generateShortId();
+  const { title, content } = validatePost(args);
+
+  if (!title) {
+    throw new ValidationError('Title can not be an empty string!');
+  }
+
+  await (options?.entityManager || con).transaction(async (manager) => {
```

Comment:
would it not mean that if entityManager is passed its already in transaction? Can we simplify it so it only accepts EntityManager and uses that whatever it is?

Reply:
Updated ✔️